src/components/productsGrid/ProductsGrid.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  Button,
+  LoadingMessage,
+  ErrorMessage,
+  HasNoResultsMessage,
+} from "../ui";
+import { ProductCard } from "./ProductCard";
+import { useProducts } from "../../hooks/useProducts";
+import { Product } from "../../types";
+import { useState } from "react";
+
+type ProductsGridProps = {
+  searchTerm: string;
+  sortBy: string;
+  sortOrder: string;
+};
+
+export const ProductsGrid = ({
+  searchTerm,
+  sortBy,
+  sortOrder,
+}: ProductsGridProps) => {
+  const [limit, setLimit] = useState(6);
+  const [isLoadingMore, setIsLoadingMore] = useState(false);
+  const {
+    data = [],
```

Comment:
why this? if you only leave it as 
```
const {
data,
...
```

you can then check` if (!data) return <HasNoResultsMessage />;` instead of using the `hasResults`